app/src/main/java/net/osmtracker/gpx/ExportToStorageTask.java

```diff
@@ -86,32 +88,33 @@ private boolean isExternalStorageWritable() {
 	}
 
 	// Create before returning if not exists
-    public File getBaseExportDirectory(SharedPreferences prefs) throws ExportTrackException {
-
-		if (!isExternalStorageWritable()) {
+	public File getBaseExportDirectory(SharedPreferences prefs) throws ExportTrackException {
+		// Verify if folder exists and is writable
+		if (!Environment.MEDIA_MOUNTED.equals(Environment.getExternalStorageState())) {
 			throw new ExportTrackException(
```

Comment:
Nope. You should read code of function isExternalStorageWritable. This function already do the job.

Reply:
You're right! I reviewed the isExternalStorageWritable function, and it already handles the logic for ensuring that the storage is writable. Thanks for pointing that out, I’ll make sure the implementation reflects that.